### PR TITLE
Pre-commit: Update clang-format to 18.1.8

### PR DESCRIPTION
**Pull request by akien-mga**
This requires updating `pre-commit` and `identify` Python packages to their latest versions, to add support for the `metal` file type:

```
pip install --upgrade pre-commit identify
```
(Technically only updating `identify` might be enough but it's good to be up-to-date for good measure.)

Also update ruff pre-commit to 0.6.5, doesn't seem to cause changes.

---

The need to update `identify` might be a bit annoying as users will run into cryptic errors like this:
```
An error has occurred: InvalidManifestError: 
==> File /home/akien/.cache/pre-commit/repolyo4sjx6/.pre-commit-hooks.yaml
==> At Hook(id='clang-format')
==> At key: types_or
==> At index 9
=====> Type tag 'textproto' is not recognized.  Try upgrading identify and pre-commit?
Check the log at /home/akien/.cache/pre-commit/pre-commit.log
```

I wonder if we can have a pre-commit hook that runs first and makes sure `identify` is up to date?

### CODE REVIEW

File: drivers/gles3/storage/render_scene_buffers_gles3.h [hunk 1]
```diff
@@ -103,9 +103,9 @@ class RenderSceneBuffersGLES3 : public RenderSceneBuffers {
 	virtual void configure(const RenderSceneBuffersConfiguration *p_config) override;
 	void configure_for_probe(Size2i p_size);
 
-	virtual void set_fsr_sharpness(float p_fsr_sharpness) override{};
-	virtual void set_texture_mipmap_bias(float p_texture_mipmap_bias) override{};
-	virtual void set_use_debanding(bool p_use_debanding) override{};
+	virtual void set_fsr_sharpness(float p_fsr_sharpness) override {}
```
Comment (akien-mga) [inline drivers/gles3/storage/render_scene_buffers_gles3.h:106]: clang-format just fixed the missing space, but I took the opportunity to remove the unnecessary semicolon.

Comment (akien-mga) [reply drivers/gles3/storage/render_scene_buffers_gles3.h:106]: BTW if anyone wants a challenge and try to finally write a script or find a clang-tidy hook that can remove useless semicolons, be my guest :D 
I've tried and failed a few times, but I get annoyed every time I find some in the codebase.

Comment (Repiteo) [reply drivers/gles3/storage/render_scene_buffers_gles3.h:106]: I dunno about clang-tidy, but clang-format certainly has unnecessary semicolon checks. The ramifications of that are pretty wide-reaching, but I can probably make a draft PR for it

**Review by Repiteo — APPROVED** [review]
Seems to largely supersede #93313, the only real exception being `.clang-format` itself but that can just become its own PR

**Comment by akien-mga** [comment]
Oh my bad, I totally forgot about this one. We can merge yours after rebase instead, updating the `.clang-format` file would be great indeed. I'll review it again tomorrow.

**Comment by akien-mga** [comment]
Duplicate of #93313.

**Pull request closed without merging**